### PR TITLE
feat: unify workflows and add ability to manually trigger CI image

### PR DESCRIPTION
**Pull request by tustvold**
After merging a PR that updates the CI image it would be helpful to be able to trigger recompilation of that image. CircleCI does not appear to have a `changes` filter where it runs a workflow if a given path has been modified, so the simplest solution is just to have the ability for a user to manually trigger a pipeline run with a parameter to force a new CI image.

This PR adds this functionality. It also merges the perf_image run into a single CI job, so that we aren't running two workflows on merge to main for no apparent reason. This also avoids an issue with the current setup where certain job failures wouldn't prevent the image from being published :scream: 

I followed these docs to write this:

- https://circleci.com/docs/2.0/workflows/
- https://circleci.com/docs/2.0/configuration-reference/#logic-statements
- https://support.circleci.com/hc/en-us/articles/360050351292-How-to-trigger-a-workflow-via-CircleCI-API-v2

### CODE REVIEW

**Comment by tustvold** [comment]
Example of a manually triggered pipeline - https://app.circleci.com/pipelines/github/influxdata/influxdb_iox/12453/workflows/07dd274c-541c-4f92-9ea5-d95c668d48f4

**Review by alamb — COMMENTED** [review]
Can you please update relevant docs about how to use this new flow?

> ability for a user to manually trigger a pipeline run with a parameter to force a new CI image.

Specifically, how would I manually test this run?

Also, do you have any suggestions on how to test this PR?

**Comment by tustvold** [comment]
@alamb I added some docs PTAL

File: .circleci/config.yml [hunk 1]
```diff
@@ -7,20 +7,25 @@
...
-# CI for all other branches:
+# On main if all these checks pass it will then additionally compile in "release" mode and
+# publish a docker image to quay.io/influxdb/fusion:$COMMIT_SHA
+#
+# Manual CI Image:
+#
+# It is possible to manually trigger a rebuild of the image used in CI. To do this, navigate to
+# https://app.circleci.com/pipelines/github/influxdata/influxdb_iox?branch=main (overriding the
+# branch name if desired). Then:
+# - Click "Run Pipeline" in the top-right
```
Comment (alamb) [inline .circleci/config.yml:22]: I tried to follow these directions but my pipeline failed: https://app.circleci.com/pipelines/github/influxdata/influxdb_iox/12472/workflows/377a56e3-cb5c-4692-b5ed-cd6301469dee/jobs/90188

```
https://app.circleci.com/pipelines/github/influxdata/influxdb_iox/12472/workflows/377a56e3-cb5c-4692-b5ed-cd6301469dee/jobs/90188
```

🤔

Comment (alamb) [reply .circleci/config.yml:22]: AH, but perhaps that is because main doesn't have this change yet 💡 !

Comment (alamb) [reply .circleci/config.yml:22]: 🤔  no that did not seem to work either: https://app.circleci.com/pipelines/github/influxdata/influxdb_iox/12475/workflows/85b0f85f-3931-4742-962a-4a02bd554e08/jobs/90218

Comment (tustvold) [reply .circleci/config.yml:22]: > Type error for argument ci_image: expected type: boolean, actual value: \"true\" (type string)

You need to set a boolean

**Comment by alamb** [comment]
Gah!

**Comment by alamb** [comment]
https://app.circleci.com/pipelines/github/influxdata/influxdb_iox/12476/workflows/3df1ffbe-6977-4a19-bad2-17ccee402d47/jobs/90219 seems much better

**Comment by tustvold** [comment]
The UX is pretty terrible, it would help if it parsed the config instead of erroring after the fact but hey ho, there didn't use to be a mechanism other than the API so I'll take it :laughing:

**Comment by alamb** [comment]
https://app.circleci.com/pipelines/github/influxdata/influxdb_iox/12476/workflows/3df1ffbe-6977-4a19-bad2-17ccee402d47/jobs/90219 passed 👍

**Review by alamb — APPROVED** [review]

**Pull request merged**